needle.pm

```diff
@@ -276,13 +276,34 @@ sub has_tag {
 }
 
 sub has_property {
-    my ($self, $tag) = @_;
+    my ($self, $pro) = @_;
     for my $t (@{$self->{properties}}) {
-        return 1 if ($t eq $tag);
+        if (ref($t) eq "HASH") {
+            return 1 if ($t->{name} eq $pro);
+        }
+        else {
+            return 1 if ($t eq $pro);
+        }
     }
     return 0;
 }
 
+sub get_property_value {
+    my ($self, $pro) = @_;
+    for my $t (@{$self->{properties}}) {
+        if (ref($t) eq "HASH") {
+            return $t->{value} if ($t->{name} eq $pro);
+        }
+        else {
+            return "" if ($pro ne "workaround");
+            if ($self->{name} =~ /\S+\-(bsc|poo)(\d+)\-\S+/) {
```

Comment:
Having this fallback inside the loop body (although it does not use `$t`) looks wrong, too. Shouldn't this only be executed if there is no matching HASH?